rotkehlchen/chain/evm/decoding/interfaces.py

```diff
@@ -170,9 +185,91 @@ def _decode_vote_cast(self, context: DecoderContext) -> DecodingOutput:
 
 
 class ReloadableDecoderMixin(metaclass=ABCMeta):
+    """Used by decoders of protocols that use data that can be reloaded from the db or from a
+    remote data source, to the decoder's memory."""
 
     @abstractmethod
     def reload_data(self) -> Optional[Mapping[ChecksumEvmAddress, tuple[Any, ...]]]:
         """Subclasses may implement this to be able to reload some of the decoder's properties
-        Returns only new mappings of addresses to decode functions
+        Returns only new mappings of addresses to decode functions"""
+
+
+class ReloadablePoolsAndGaugesDecoderMixin(ReloadableDecoderMixin):
+    """Used by decoders of protocols that have pools and gauges stored in the globaldb cache
+    tables. It can reload them to the decoder's memory.
+    """
+
+    def __init__(
+            self,
+            evm_inquirer: 'EvmNodeInquirer',
+            cache_type_to_check_for_freshness: CacheType,
+            query_data_method:
+            Callable[['OptimismInquirer'], Optional[list['VelodromePoolData']]] |
```

Comment:
As mentioned in discord using `|` is incosistent with the rest of the codebase. Though we will switch to that notation soon.